prez/services/annotations.py

```diff
@@ -41,6 +44,44 @@ async def get_annotations(terms_and_dtypes: Set[URIRef], repo: Repo, system_repo
     return annotations_g
 
 
+async def get_annotations_for_oxigraph(terms_and_dtypes: Set[OxiNamedNode], repo: Repo, system_repo: Repo) -> OxiStore:
+    """
+    This function processes the terms and their data types. It first retrieves the cached results for the given terms
+    and data types. Then, it processes the terms that are not cached. The results are added to a graph which is then
+    returned.
+
+    Args:
+        terms_and_dtypes (set): A list of tuples where each tuple contains a term and its data type.
+        repo (Repo): An instance of the Repo class.
+        system_repo (Repo): An instance of the Repo class with the Prez system graph.
+
+    Returns:
+        annotations_store (OxiStore): A oxigraph store containing the processed terms and their data types.
+    """
+    annotations_store = OxiStore()
+    oxi_default_graph = OxiDefaultGraph()
+    cache = caches.get("default")  # This always returns the SAME instance
+    # Cache always uses URIRefs, so we convert OxiNamedNode to URIRef
+    results = await cache.multi_get(list(URIRef(t.value) for t in terms_and_dtypes))
```

Comment:
The aiocache uses the pickle serializer, to store the mapping of `URIRef` to `frozenset((p,o))` where p and o are RDFLib terms. We cannot have the equivelent separate cache for mapping Oxigraph `NamedNode` to `frozenset((p,o))` where p and o are Oxigraph terms. Because Oxigraph terms are not picklable.

So to use the annotations cache, we need to convert the Oxigraph terms to RDFlib terms. And when loading from the cache, we convert them back again (see below).

One upside to doing it this way is it means `get_annotations_for_oxigraph` and `get_annotations` functions called from different codepaths can both use and benefit from the same annotations cache.